drivers/flash/flash_stm32_xspi.c

```diff
@@ -2444,9 +2448,19 @@ static struct flash_stm32_xspi_data flash_stm32_xspi_dev_data = {
 					: HAL_XSPI_CSSEL_NCS2),
 #endif
 			.FreeRunningClock = HAL_XSPI_FREERUNCLK_DISABLE,
-#if defined(OCTOSPI_DCR4_REFRESH)
+#if defined(OCTOSPI_DCR1_DLYBYP) || defined(XSPI_DCR1_DLYBYP)
+#if STM32_XSPI_DLYB_BYPASSED
+			.DelayBlockBypass = HAL_XSPI_DELAY_BLOCK_BYPASS;
+#else
+			.DelayBlockBypass = HAL_XSPI_DELAY_BLOCK_ON;
+#endif /* STM32_XSPI_DLYB_BYPASSED */
```

Comment:
Use of `;` isn't correct.

Also let's align syntax on other properties above
```suggestion
			.DelayBlockBypass = (DT_PROP(STM32_XSPI_NODE, dlyb_bypass)
					? HAL_XSPI_DELAY_BLOCK_BYPASS
					: HAL_XSPI_DELAY_BLOCK_ON),
```
And `STM32_XSPI_DLYB_BYPASSED` definition can be removed.